### PR TITLE
Add Google Cloud Logging (GCL) sink for events to heapster

### PR DESCRIPTION
**Pull request by saad-ali**
Screenshot of events in GCL:
gcl

### CODE REVIEW

**Comment by saad-ali** [comment]
CC @vishh

File: sinks/gcl/auth.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/gcloud-golang/compute/metadata"
+)
+
+type authToken struct {
```
Comment (vmarmol) [inline sinks/gcl/auth.go]: We should consider consolidating these two auth implementations (GCM, GCL) into util in Heapster

Comment (vishh) [inline sinks/gcl/auth.go]: +1

File: sinks/gcl/driver.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+	err = impl.refreshToken()
+	if err != nil {
+		return nil, err
+	}
+
+	return impl, nil
+}
+
+func (sink *gclSink) DebugInfo() string {
+	return "Sink Type: Google Cloud Logging (GCL)"
```
Comment (vishh) [inline sinks/gcl/driver.go]: We should surface write errors here to aid in debugging.

File: sinks/gcl/driver.go [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+	UserId string `json:"userId,omitempty"`
+
+	// A set of (key, value) data that provides additional information about the log entry.
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+const (
+	EventLoggingSeverity     = "NOTICE"
+	EventLoggingServiceName  = "custom.googleapis.com"
+	EventLoggingLogName      = "KubernetesEventsLog"
```
Comment (vishh) [inline sinks/gcl/driver.go]: Should this be "kubernetes.io" to match the naming for GCM?

File: sinks/gcl/driver.go [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+		return err
+	}
+
+	defer resp.Body.Close()
+	out, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != 200 {
```
Comment (vishh) [inline sinks/gcl/driver.go]: nit: Use http.StatusOK instead

**Comment by vishh** [comment]
LGTM excepting the few comments. I will let @vmarmol do a final review and merge this PR!

**Comment by vishh** [comment]
Nice job @saad-ali!

**Comment by saad-ali** [comment]
PTAL

**Comment by vishh** [comment]
Ok merging this PR! @vmarmol if you have any comments, we can address them in a subsequent PR.

**Pull request merged**